app/coffee/Features/Project/ProjectApiController.coffee

```diff
@@ -7,8 +8,11 @@ module.exports =
 	getProjectDetails : (req, res)->
 		{project_id} = req.params
 		ProjectDetailsHandler.getDetails project_id, (err, projDetails)->
-			if err?
+			if err? and err instanceof Errors.NotFoundError
```

Comment:
This entire error block can just be passed to next(err). Express is set up to handle NotFoundErrors with a 404, or a 500 for other errors.